site/zenodo_rdm/legacy/requests/utils.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# ZenodoRDM is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Utility functions."""
+from datetime import datetime, timedelta
+
+from invenio_access.permissions import system_identity
+from invenio_records.dictutils import dict_lookup
+from invenio_records_resources.services.uow import unit_of_work
+from invenio_requests import current_request_type_registry, current_requests_service
+from invenio_requests.resolvers.registry import ResolverRegistry
+from invenio_search.engine import dsl
+from pytz import utc
+
+from .record_upgrade import LegacyRecordUpgrade
+
+
+def exists(identity, record):
+    """Return the request id if an open request already exists, else None."""
+    results = current_requests_service.search(
+        identity,
+        extra_filter=dsl.query.Bool(
+            "must",
+            must=[
+                dsl.Q("term", **{"topic.record": record.pid.pid_value}),
+                dsl.Q("term", **{"type": LegacyRecordUpgrade.type_id}),
+                dsl.Q("term", **{"is_open": True}),
+            ],
+        ),
+    )
+    return next(results.hits)["id"] if results.total > 0 else None
+
+
+@unit_of_work()
+def submit_record_upgrade_request(record, uow, comment=None):
```

Comment:
This is handy, but it might be too slow when Zenodo will migrate data (one ES check per record). They might want to handle the sending of notifications (e-mails) differently.
@slint can you please comment on this?